.travis.yml

```diff
@@ -0,0 +1,47 @@
+dist: bionic
+language: cpp
+
+jobs:
+  include:
+  - os: linux
+    compiler: gcc
+    dist: bionic
+    addons:
+      apt:
+        packages:
+        - ccache
+        - cmake
+        - clang-format
+        - libgmp3-dev
+        - parallel
+    before_install:
+      - .travis/preinstall.sh
+    script:
+      - cd $TRAVIS_BUILD_DIR && mkdir build && cd build && cmake .. && make
```

Comment:
I notice you do not use any `-j` or `--jobs` options with `make` where you have `make` in `.travis.yml` and `.travis/preinstall.sh`. I am not really that experienced with Travis CI, but would using some parallelism such as `make -j4` enable these scripts to run faster?

Reply:
Yes, we could add that. I just initially didn't add it for debug. Will add it.